scipy/optimize/_optimize.py

```diff
@@ -3544,9 +3544,10 @@ def brute(func, ranges, args=(), Ns=20, full_output=0, finish=fmin,
             success = res[-1] == 0
         if not success:
             if disp:
-                print("Warning: Either final optimization did not succeed "
-                      "or `finish` does not return `statuscode` as its last "
-                      "argument.")
+                warnings.warn(
+                    "Either final optimization did not succeed "
+                    "or `finish` does not return `statuscode` as its last "
+                    "argument.", RuntimeWarning)
```

Comment:
This one should be 2
```suggestion
                    "argument.", RuntimeWarning, 2)
```